hack/tools/release/weekly/main.go

```diff
@@ -50,47 +53,63 @@ var (
 		release.Unknown,
 	}
 
-	from = flag.String("from", "", "Include commits starting from and including this date. Accepts format: YYYY-MM-DD")
-	to   = flag.String("to", "", "Include commits up to and including this date. Accepts format: YYYY-MM-DD")
+	since   string
+	until   string
+	branch  string
+	ghToken string
 
-	milestone = flag.String("milestone", "v1.4", "Milestone. Accepts format: v1.4")
+	timeLayout = "2006-01-02"
 
 	tagRegex = regexp.MustCompile(`^\[release-[\w-\.]*\]`)
 )
 
 func main() {
+	flag.StringVar(&since, "since", "", "Include commits starting from and including this date. Accepts format: YYYY-MM-DD")
+	flag.StringVar(&until, "until", "", "Include commits up to and including this date. Accepts format: YYYY-MM-DD")
+	flag.StringVar(&branch, "branch", "release-1.6", "Release branch. Accepts formats: main, release-1.6")
+	flag.StringVar(&ghToken, "gh-token", "", "Github token. Enables a higher api request rate.")
 	flag.Parse()
 	os.Exit(run())
 }
 
-// Since git doesn't include the last day in rev-list we want to increase 1 day to include it in the interval.
-func increaseDateByOneDay(date string) (string, error) {
-	layout := "2006-01-02"
-	datetime, err := time.Parse(layout, date)
+func run() int {
+	if since == "" && until == "" {
+		fmt.Println("--since and --until are required together or both unset")
+		return 1
+	}
+
+	client := createGitHubClient(ghToken)
+	repo := "cluster-api"
+	owner := "kubernetes-sigs"
```

Comment:
Can these be moved to the top of the file as constants? I don't think these are set to anything different throughout the file.